sdk/daml-lf/validation/src/main/scala/com/daml/lf/validation/Upgrading.scala

```diff
@@ -321,10 +354,28 @@ case class TypecheckUpgrades(packagesAndIds: Upgrading[(Ref.PackageId, Ast.Packa
           _package.map(_.modules),
           (name: Ref.ModuleName, _: Ast.Module) => UpgradeError.MissingModule(name),
         )
+      _ <- checkDeps()
       _ <- tryAll(upgradedModules.values, checkModule(_))
     } yield ()
   }
 
+  private def checkDeps(): Try[Unit] = {
+    val (_new @ _, existing, _deleted @ _) = extractDelExistNew(dependencies.map(_.values.toMap))
+    tryAll(existing, checkDep).map(_ => ())
+  }
+
+  private def checkDep(dep: (Ref.PackageName, Upgrading[Ref.PackageVersion])): Try[Unit] = {
+    val (depName, depVersions) = dep
+    failIf(
+      depVersions.present < depVersions.past,
```

Comment:
There already is a "comparable" instance for Ref.PackageVersion that does the correct thing, so we don't need any utils like we did with Haskell